changes/tasks.py

```diff
@@ -572,14 +572,37 @@ def switch_channel(self, change):
                 )
 
             elif change.data["channel"] == "sms" and "whatsapp" in short_name:
-                # Change any WhatsApp subscriptions to SMS
-                sbm_client.update_subscription(sub["id"], {"active": False})
 
                 # There's no service info for SMS
                 if "service_info" in short_name:
                     continue
 
-                messageset = messagesets_rev[re.sub("^whatsapp_", "", short_name)]
+                messageset = messagesets_rev.get(re.sub("^whatsapp_", "", short_name))
+
+                if not messageset:
```

Comment:
I'm worried that we're including too much here. It would rather be better that if we forget about any messageset that's whatsapp only, that it results in an exception, rather than sending the message that is very specific for 1-2 messaging, and us not noticing that anything is wrong.

So I think we should rather check if the message is the specific messageset that we're concerned about here, rather than checking for any missing messageset.